src/Menu/PanelMenu.tsx

```diff
@@ -38,50 +42,32 @@ export class PanelMenu extends SceneObjectBase<PanelMenuState> implements VizPan
   constructor(state: Partial<PanelMenuState>) {
     super({ ...state, addExplorationsLink: state.addExplorationsLink ?? true });
     this.addActivationHandler(() => {
-      let exploreUrl: Promise<string | undefined> | undefined;
-      try {
-        const viz = sceneGraph.getAncestor(this, VizPanel);
-        const panelData = sceneGraph.getData(viz).state.data;
-        if (!panelData) {
-          throw new Error('Cannot get link to explore, no panel data found');
-        }
-        const queryRunner = getQueryRunnerFor(viz);
-        const queries = queryRunner?.state.queries ?? [];
-        queries.forEach((query) => {
-          // removing legendFormat to get verbose legend in Explore
-          delete query.legendFormat;
-        });
-        // 'this' scene object contain the variable for the metric name which is correctly interpolated into the explore url
-        // when used in the metric select scene case,
-        // this will get the explore url with interpolated variables and include the labels __ignore_usage__, this is a known issue
-        // in the metric scene we do not get use the __ignore_usage__ labels in the explore url
-        exploreUrl = getExploreURL(panelData, this, panelData.timeRange, (query) => {
-          // remove __ignore_usage__="" from the query
-          if ('expr' in query && typeof query.expr === 'string' && query.expr.includes('__ignore_usage__')) {
-            return {
-              ...query,
-              expr: query.expr.replace(/,?__ignore_usage__="",?/, ''), // also remove leading/trailing comma if present
-            };
-          }
-
-          return query;
-        });
-      } catch {}
+      // Check if this is the main metric graph panel
+      // The main panel is created with labelName matching the metric in MetricGraphScene
+      const trail = getTrailFor(this);
+      const isMainGraphPanel = this.state.labelName === trail.state.metric;
```

Comment:
This is very brittle.

Maybe the easiest and most resilient thing to do instead would be to pass a key to the panel menu [when it's created in MetricGraphScene](https://github.com/grafana/metrics-drilldown/blob/main/src/MetricGraphScene.tsx#L90)? And then, this check would work:

```suggestion
      const isMainGraphPanel = this.state.key === TOPWVIEW_PANEL_MENU_KEY;
```

Reply:
Also, a very simple readability improvement:

```ts
 this.addActivationHandler(this.onActivate.bind(this));
```